drivers/Serial.h

```diff
@@ -81,6 +81,23 @@ class Serial : public SerialBase, public Stream {
      */
     Serial(PinName tx, PinName rx, int baud);
 
+    /* Stream gives us a FileHandle with non-functional poll()/readable()/writable. Pass through
+     * the calls from the SerialBase instead for backwards compatibility. This problem is
+     * part of why Stream and Serial should be deprecated.
+     */
+    bool readable()
```

Comment:
Actually, no, that doesn't really work out.

The issue here is that SerialBase has `int readable()` and Stream has non-functioning `bool readable() const`.

Attempting to call `readable()` on a Serial would cause an ambiguity error. Adding any definition of readable in Serial, whether const or not, shadows both of those, solving the ambiguity. As we are passing through to SerialBase's non-const method, then Serial's readable is non-const.

SerialBase's readable is non-const because it calls lock() - I see consensus that built-in locks can be mutable, but not on constness of lock methods in combined function+lock objects. Because overriding lock() is part of SerialBase's API, I don't think it could be easily changed anyway.

I don't particularly fancy the const_cast, and there is no need for this to be const to shadow FileHandle::readable() as intended.